modules/storage-rocksdb/src/main/java/org/apache/ignite/internal/storage/rocksdb/RocksDbMvPartitionStorage.java

```diff
@@ -570,17 +589,26 @@ public void commitWrite(RowId rowId, HybridTimestamp timestamp) throws StorageEx
                 byte[] valueBytes = writeBatch.getFromBatchAndDB(db, cf, readOpts, uncommittedKeyBytes);
 
                 if (valueBytes == null) {
-                    //the chain doesn't contain an uncommitted write intent
+                    // The chain doesn't contain an uncommitted write intent.
                     return null;
                 }
 
+                boolean isNewValueTombstone = valueBytes.length == VALUE_HEADER_SIZE;
+
+                // Both this and previous values for the row id are tombstones.
+                boolean newAndPrevTombstones = maybeAddToGcQueue(writeBatch, rowId, timestamp, isNewValueTombstone);
+
                 // Delete pending write.
                 writeBatch.delete(cf, uncommittedKeyBytes);
 
-                // Add timestamp to the key, and put the value back into the storage.
-                putTimestamp(keyBuf, timestamp);
+                // We only write tombstone if the previous value for the same row id was not a tombstone.
+                // So there won't be consecutive tombstones for the same row id.
+                if (!newAndPrevTombstones) {
```

Comment:
Why wasn't this check present in the previous implementation? Was it a bug? And when can we have two tombstones for the same key anyway?